src/test/java/org/example/dao/MySqlJobRoleDaoTest.java

```diff
@@ -0,0 +1,58 @@
+package org.example.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.when;
+
+import java.io.FileReader;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.List;
+import org.example.daos.MySqlJobRoleDao;
+import org.example.daos.interfaces.JobRoleDao;
+import org.example.exception.JobRoleDaoException;
+import org.example.models.JobRole;
+import org.example.utils.DatabaseConnector;
+import org.h2.tools.RunScript;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class MySqlJobRoleDaoTest {
+    private Connection h2Connection;
+    private DatabaseConnector mockDatabaseConnector;
+    private JobRoleDao jobRoleDao;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        h2Connection = DriverManager.getConnection("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1");
+        RunScript.execute(h2Connection, new FileReader("src/test/resources/schema.sql"));
+        RunScript.execute(h2Connection, new FileReader("src/test/resources/data.sql"));
+
+        mockDatabaseConnector = Mockito.mock(DatabaseConnector.class);
+        when(mockDatabaseConnector.getConnection()).thenReturn(h2Connection);
+
+        jobRoleDao = new MySqlJobRoleDao(mockDatabaseConnector);
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        h2Connection.close();
+    }
+
+    @Test
+    public void getJobRoles_shouldReturnListOfJobRoles_whenDatabaseReturnsRowsOfJobRoles() throws JobRoleDaoException {
```

Comment:
are there any more tests you want to write here?

Reply:
Ah yes, to test if the db returns empty rows

Reply:
Added